src/select/expr_term.rs

```diff
@@ -0,0 +1,290 @@
+use crate::select::cmp::*;
+use crate::select::select_value::{SelectValue, SelectValueType};
+use crate::select::{to_f64, FilterKey};
+
+#[derive(Debug, PartialEq)]
+pub(super) enum ExprTerm<'a, T>
+where
+    T: SelectValue,
+{
+    String(String),
+    Long(i64),
+    Double(f64),
```

Comment:
The original has just `Number`, what was the reason for splitting into `Long`/`Double`?

Reply:
Number itself was split (IIRC) so I thought it redundant to create another enum for number that by itself will be split...